state/azure/cosmosdb/cosmosdb.go

```diff
@@ -287,47 +245,33 @@ func (c *StateStore) Set(req *state.SetRequest) error {
 	}
 
 	partitionKey := populatePartitionMetadata(req.Key, req.Metadata)
-	options := []documentdb.CallOption{documentdb.PartitionKey(partitionKey)}
+	options := azcosmos.ItemOptions{}
 
 	if req.ETag != nil {
-		options = append(options, documentdb.IfMatch(*req.ETag))
+		etag := azcore.ETag(*req.ETag)
+		options.IfMatchEtag = &etag
 	}
 	if req.Options.Concurrency == state.FirstWrite && (req.ETag == nil || *req.ETag == "") {
-		etag := uuid.NewString()
-		options = append(options, documentdb.IfMatch(etag))
+		newTag := azcore.ETag(uuid.NewString())
+		options.IfMatchEtag = &newTag
 	}
 	if req.Options.Consistency == state.Strong {
-		options = append(options, documentdb.ConsistencyLevel(documentdb.Strong))
+		options.ConsistencyLevel = azcosmos.ConsistencyLevelStrong.ToPtr()
 	}
 	if req.Options.Consistency == state.Eventual {
-		options = append(options, documentdb.ConsistencyLevel(documentdb.Eventual))
+		options.ConsistencyLevel = azcosmos.ConsistencyLevelEventual.ToPtr()
 	}
 
-	doc, err := createUpsertItem(c.contentType, *req, partitionKey)
+	marshalled, err := json.Marshal(req.Value)
 	if err != nil {
 		return err
 	}
-	err = retryOperation(func() error {
-		_, innerErr := c.client.UpsertDocument(c.getCollectionLink(), &doc, options...)
-		if innerErr != nil {
-			if isTooManyRequestsError(innerErr) {
-				return innerErr
-			}
-			return backoff.Permanent(innerErr)
-		}
-		return nil
-	}, func(err error, d time.Duration) {
-		c.logger.Warnf("CosmosDB state store Set request failed: %v; retrying in %s", err, d)
-	}, 20*time.Second)
 
+	pk := azcosmos.NewPartitionKeyString(partitionKey)
+	_, err = c.client.UpsertItem(context.Background(), pk, marshalled, &options)
```

Comment:
Please make sure to add a timeout (using the defaultTimeout constant) to the context